### PR TITLE
Refactor text util surrogate helpers

### PR DESCRIPTION
**Pull request by dcodeIO**
Refactors surrogate code point helpers to text util for reuse.

- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

File: src/tokenizer.ts [hunk 1]
```diff
@@ -1739,12 +1741,7 @@ export class Tokenizer extends DiagnosticEmitter {
         ? text.substring(startIfTaggedTemplate, this.pos)
         : "";
     }
-    return value32 < 0x10000
-      ? String.fromCharCode(value32)
-      : String.fromCharCode(
-        ((value32 - 0x10000) >>> 10) | 0xD800,
-        ((value32 - 0x10000) & 1023) | 0xDC00
-      );
```
Comment (MaxGraey) [inline src/tokenizer.ts:1747]: I guess this manual routine should be faster than `String.fromCodePoint`. Did you measured this?

File: src/util/text.ts [hunk 2]
```diff
@@ -175,6 +175,50 @@ export function isWhiteSpace(c: i32): bool {
...
+
+/** Tests if a code unit or code point is a low (trail) surrogate. */
+export function isLowSurrogate(c: i32): bool {
+  // FC00: 11111 1 0000000000 Mask
+  // DC00: 11011 1 XXXXXXXXXX Low/Trail surrogate
+  return (c & 0xFC00) == SURROGATE_LOW;
+}
+
+/** Converts a surrogate pair to its respective code point. */
+export function surrogatesToCodepoint(hi: i32, lo: i32): i32 {
```
Comment (MaxGraey) [inline src/util/text.ts:218]: How about `combineSurrogates` ?

File: src/tokenizer.ts (hunk 1 above)
Comment (dcodeIO) [reply src/tokenizer.ts:1747]: I did not measure it, but in an ideal world we should just be able to use `fromCodePoint` where the input is a code point.

Comment (MaxGraey) [reply src/tokenizer.ts:1747]: I've just seen somewhere that this manual approach is slightly faster than the built-in fromCodePoint

Comment (dcodeIO) [reply src/tokenizer.ts:1747]: OTH it's less redundant code, and to the point. Should be fine I think.

Comment (MaxGraey) [reply src/tokenizer.ts:1747]: Hmm, on Chrome this nearly the same now. On FF native codePoint faster by 1.5x
https://jsben.ch/sm8Tt

File: src/module.ts [hunk 3]
```diff
@@ -10,6 +10,7 @@
 
 import { BuiltinNames } from "./builtins";
 import { Target } from "./common";
+import { isHighSurrogate, isLowSurrogate, combineSurrogates, SURROGATE_HIGH, SURROGATE_LOW } from "./util";
```
Comment (MaxGraey) [inline src/module.ts:13]:
```suggestion
import { 
  isHighSurrogate, 
  isLowSurrogate, 
  combineSurrogates, 
  SURROGATE_HIGH, 
  SURROGATE_LOW 
} from "./util";
```

**Review by MaxGraey — APPROVED** [review]

**Pull request merged**